src/lib.rs

```diff
@@ -193,6 +198,18 @@ pub struct Client<T: Runtime> {
     page_size: u32,
 }
 
+/// Construction options for a signed extrinsic
+#[derive(Clone)]
+pub struct SignedOptions {
+    /// The period, measured in blocks, that transaction will live for, starting from a checkpoint
+    /// block. A good default is 64 (64 * 6secs = 6min 40sec).
+    ///
+    /// `era_period == None`: immortal transaction.
+    /// `0 <= era_period <= 65536`: rounded up to the closest power of 2, starting at 4.
+    /// `65536 < era_period`: 65536.
```

Comment:
last line means: min(era_period, 65536)

Reply:
Also, can you explain why `era period` is capped at `65536`?

Reply:
That number came from my interpretation of these comments: https://github.com/paritytech/substrate/blob/dbf2163250833e6ac898e7f6c3c8f89f08a7c19d/primitives/runtime/src/generic/era.rs#L50-L65

Reply:
Just learned from @joepetrowski and wanted to clarify - this number is the maximum that can be encoded, but `BlockHashCount` is the maximum amount of blocks that the transaction is actually valid for (on Polkadot and Kusama that is 2400)

Reply:
TIL!
Seems like there's a lot more complexity here than I first thought. Would be good to capture this in the comment (and perhaps upstream as well?).

Reply:
It's a funny nuance. It's not that the transaction is invalid per se, it's that the runtime stores `BlockHashCount` blocks in state as reference. If the runtime doesn't have the transaction's checkpoint block to reference against, it's _unable_ to say whether the block is valid or not, so just rejects it on those grounds.

Reply:
Alright, that's annoying for this usecase. 

If I understand this correctly `BlockHashCount` is FRAME-related and configured by each runtime. Thus, if I want to use my custom runtime with `BlockHashCount == 3` and `era_period > 3` then it will be rejected and would hard to figure out as a user of `substrate-subxt`.

Next question is: can we fetch `BlockHashCount` in the `Client constructor/builder` via RPC [similary as other runtime stuff](https://github.com/paritytech/substrate-subxt/blob/master/src/lib.rs#L165-#L171) then do `min(block_hash_count, mortality_period)`?

Reply:
Yes, it's available over RPC: https://github.com/paritytech/substrate/blob/31499cd29ed30df932fb71b7459796f7160d0272/frame/system/src/lib.rs#L508